### ISSUE TITLE
Map Hats / Headbands to All Frames + Add Repals Where Needed

### ISSUE THREAD

**Issue by jrconway3** [sanderfrenken/Universal-LPC-Spritesheet-Character-Generator#206]
So I was looking over this to see how hard it would be to finish these up. Hats/Headbands tend to be fairly easy, as is anything on the head. As long as I have cutouts we're good.

The biggest issue here is just how MUCH there is, and as it turns out, there's way more than I anticipated. Let's break everything down.

NOTE: I will not be doing new repalettes if the palettes of the core color is more than 7-8 colors, primarily 7. I would have to come up with new fill-in colors for every color above that level and that would take a long time.

Head Coverings:
1. Hijab (very easy repal, might need a male/female split due to the hanging strands, but probably isn't too necessary)
2. Hoods (if I ignore the partial transparency and the black, there's about 10 colors that vary from each; this is really annoying, but could be doable if I add more colors to other palettes later... not worth it right now, though, I'll just stick with the current three, though I may implement a way to just repal between the three sets; might need a male/female split)
3. Bandana (repal shouldn't be too hard, but the white is really weird; the white has a lot of stray extra colors that are utterly pointless because you can't even tell the difference)
4. Tied Head Band (I REALLY like this one and it already has a full list of repals)

Common Hats:
1. Christmas Elf (this appears to have both the trim and main hat with ALL color variations; if I do this I'm going to split them into two separate layers)
2. Santa Hat (similar to Christmas Elf, but only white for the trim; I could add repals for the trim as well for fun)
3. Feather Cap (I'm not doing recolors for this right now)
5. Crown (there's four base colors with 7 palettes, but I'll add recolors for metal recolors and all "cloth" recolors)
6. Tiara (this should be fine to easily do with repals; there's no male version, but that's an easy addition, we probably don't even need two separate versions at all here)
7. Formal Hat (I think the Brown formal hat is incorrectly layered; it looks like the two hats are the same hat, but the Brown hat got layered over itself once so it kinda got duplicated a bit)
8. Formal Round Hat (this one should be easy to drop repalettes for)

Magic Hats:
1. Standard Hat (this has existing colors in gray/red/teal/yellow/black/brown; all are the same style, but they're not easy to repal straight into other colors; perhaps I can take the "band" and repal that as a separate layer)
2. Starry Wizard Hat (includes both moon and non-moon; I can't think of a good way to easily repal this, so just map to all frames)
3. "Brown 2" Hat (this hat is terrible looking, but its already here so I'll map to all frames)

Pirate Hats:
This list is absolutely massive, I didn't expect that many. All of them already have repals, so need to keep that in mind. Most have a female/male split, but as they're on top of the head, that split is probably no longer necessary in these cases.

1. Pirate Bandana (looks similar to the Bandana above, but FAR better and far cleaner)
2. Skull Bandana (same as the Pirate Bandana, but with a skull; I could just have the skull be a separate layer with its own repals for fun, that's one option)
3. Bicorne Athwart (includes Admiral, Admiral Cockade, Commodore, and Pirate variants)
4. Bicorne Foreaft (also includes Commodore)
5. Bonnie (and Feather)
6. Cavalier (and Feather)
7. Kerchief
8. Tricorne (also includes Captain, Pirate Captain, Skull Pirate Captain, Lieutenant, Stiched, and Thatched Variations)

Reptile Hats:
This is weird. Basically, these assets are designed for Reptile heads rather than normal heads. I already have all of these assets completed, except for the hood and bandana. I'll test and see if we even need these separately or if the existing assets will work. I'm pretty sure I already tested the horns on Reptile anyway so they should look good I think...

1. Red Bandana
9. Fin (three colors)
10. Short Fin (three colors)
11. Hood
12. Horns (same was under Horns)

Even if the Fins/Horns don't quite work for Reptile, I'll still move them out of here and over to the appendages section, just with a different entry for "Reptile" that has the corrected positioning.

This list is... really intimidating. They should each individually be quite easy, and tbh, I doubt they need any cutouts either because the hands would be so low. However, its such a MASSIVE list that it will take a long time to get through.

**Comment by jrconway3** [comment]
Okay, I know how I can repal the hood. For the hood, I just take the base brown color and only repal the browns. Then I keep the original white and original black as separate options. That should work I think.

I'm going to try and get out a quick update with repals for all the "cloth" items right away.

**Comment by jrconway3** [comment]
Hood is a bit harder than I thought, though. It does require some adjustments around the shoulders.

**Comment by jrconway3** [comment]
Done so far:
1. Tied Head Band
2. Bandana (w/recolors)
3. Bordered Bandana (aka Bandana white, now with its own set of recolors; I did reduce the palette a bit, but that's because the colors were so close together they weren't even noticeable anyway.)
4. Hijab (w/recolors)
5. Crown (w/recolors)

I don't really like the crown very much. In some positions, as well, I had to lower the position of the crown because it kept exiting the frame. Other cases I had to squeeze the crown height a tad by stripping some pixel height here and there. Hopefully this looks fine, but honestly, the design leaves a lot to be desired IMO.

**Comment by jrconway3** [comment]
I want to finish the Hood as well, but this one is fairly complicated. My plan is to make two sets of repals: the first is a repal to just repal between the 13 core colors of the existing three hoods. That way I can just map _one_ hood and repal between them. The other is to make a full repal of the standard cloth color set as an extra option.

I just don't want to have to recreate the extra frames around the shoulders three times. Nor do I intend on creating 13 colorsets for 30+ palettes. I'm just going to use the core 6-7 colors and repal those alone.

**Comment by jrconway3** [comment]
Feathered cap:

I mapped this to all frames, and discovered that even though there's about 12 or so color palettes for the cap, many of them are barely used and similar to other colors. As such, I have created a color reduced version of the cap that I can repal into other colors, and I'll create a separate layer repal of the feather, too.

However, I will keep the original version of the cap with the extra palettes.

**Comment by jrconway3** [comment]
Hoods:
Out of the seven core (lighter) palettes, only one of them only existed in one frame in one pixel. I merged this palette, now I have six palettes I can easily repal into other colors, while keeping the grayscale for the rest.

Feathered Cap is basically done now. Once Hoods are done, I'll open a PR for all the cloth headpieces.

**Comment by jrconway3** [comment]
Turns out there's no easy way to mesh the three sets of hoods together into a way I can just convert between the three palettes. All the partial transparency levels are different between each one, so I'm just creating each one manually by meshing them to the style of each other as I work on them. I am partially down Jump at this point while trying to catch up with White (Brown and Black are down through Jump and Emotes).

I really, really do like this hood, though. Its very nice and stylized. I have a plan I think should work for an easy repal to other colors so we'll have tons more options.

Feathered Cap is done complete with a separate "Feather" layer overlay. I just need to test and make sure it looks good.


The first release here will be the "Cloth" directory as well as the Tied Head Band. Once I submit this, I'll look into body armour as well as getting Visors out of the way quickly.

**Comment by jrconway3** [comment]
Hood updates:
I have mapped "brown" to all frames except South Backslash, South Run, and half of the West/East run animations. I am going to _attempt_ to repal the brown palettes just between the fully opaque colors and see how well that looks for the white hood. I'm not worried about black and brown, I think they'd be fine. I just worry that the white hood might look off if I don't keep those transparencies inside the hood.

If that looks good, then I'll just forgo trying to map the defaults across all frames and only map brown good.


I have finished configuring the sheets for all these assets, as well as having a separate later for Feathered Cap.

I also already finished Crown, too. Because Crown is in the same sub-list as Tiara, I quickly threw together Tiara as well. I'll just do simple repals of these, nothing more.

**Comment by jrconway3** [comment]
PR open now. Hood is complete. I didn't map the white and black hoods separately, I simply repaled the 13 non-black, partially transparent colors to the white and black versions of the base hood. For all other colors, I repaled brown to all our standard "cloth" colors using the lightest 6 brown colors, keeping all other colors the same.


The partial transparency is annoying and is the main reason I didn't redraw the hood in some places.

**Comment by jrconway3** [comment]
Feather Cap's "base" color palette is missing.

**Comment by jrconway3** [comment]
I added the missing Feather Cap in the newest PR.


I won't work on these for a while, but I'll probably do the "general" hats and the Magic Hats next time I work on hats. After that, I'll do Pirate and "Lizard" at the same time.

The Lizard hoods are slightly different from the regular hoods so maybe I can keep this as a separate asset and map it to normal heads as well. I'll test the horns with lizard head and see how it looks (I thought I already tested the regular horns anyway, though).

The bandanas appear to be lower on Lizard compared to the regular heads...

**Comment by jrconway3** [comment]
Doing the armor is so complicated I feel like I want to drop another "quick" hat asset in between lol... I can move in and take a peek at the Lizard assets next I guess, then Magic after that.

**Comment by jrconway3** [comment]
I got distracted with other things, so I think I'm going to try and put out a quick hats update over the weekend.

**Comment by jrconway3** [comment]
The formal hats are layered over one another. Some versions of the hat have bad layering where the two versions of formal hat are layered over itself in different ways, but otherwise, these two hats are identical. I'm going to fix the layering here so the hat isn't layering over itself, and then add additional recolors while keeping the original two.

The round hat, aka the Bowler Hat, is a totally separate asset of course so that will get its own section. I think I'll rename it Bowler Hat as well because that's what it is.

**Comment by jrconway3** [comment]
Oh geez. I just realized Santa and Christmas Elf are identical except for the trim. I already planned to split the trim off into a separate layer anyway... now I kinda want to just remove the trim entirely and great a trimless version, with the trim being an overlay. But I just greatly overcomplicated what should have been a simple job.

Why do I do this to myself?! :P

EDIT: Okay, the Santa Hat I'll use as the basis. The white is still bordered by red anyway. I'll just fill out the white borders with red to finish it up.

EDIT 2: So doing this isn't actually that easy. Its not identical between the two, just very, very similar. The side angles have a red border around the white, and that red border doesn't look very good if I keep it on the trimless version. Furthermore, the red border doesn't exist on the elf version, the shading is lighter in the same spot where the red exists. That said, there's only like a couple of pixels here, so its not a huge deal.

**Comment by jrconway3** [comment]
I want to finish this issue by the end of the month so I can drop a full release on OGA with all head accessory assets (facial assets and hats, except Helmets which will go in Armor). I'll tackle Reptile and Magic next and end on Pirate.


Formal Hats and Holiday Hats are now done. I guess this is my Christmas update?

Magic Hats I'm a bit worried about. I won't recolor the very muddy brown magic hat, or the blue wizard hats, the one that has the stars variant. However, the other hats are weird... they could potentially be recolored, but the recoloring is odd. I don't know how to properly standardize it yet.

**Comment by jrconway3** [comment]
I feel like I should have animated the hats at least on run, but I wasn't feeling up to it right now.

**Comment by jrconway3** [comment]
I also completed the Magic Hats, too.

**Comment by jrconway3** [comment]
Okay, so I took a look at the Reptile list:

Fins are almost identical to the other Fins. No need to worry about these, they're only SLIGHTLY off-position, but tbh, I think the regular fins look better than the Reptile Fins.

Horns are literally identical.


The only thing left is Hood. As such, I'll just finish the Hood, then immediately start working on the Pirate hats. I'll drop the Lizard Hood (and remove the entire "Reptile" category) and do the Pirate list all in one update to finish off this card.

**Comment by jrconway3** [comment]
I'm pretty sure this hood is basically a sackloth type of hood, so I'll be labeling it as that, while leaving "Reptile" in to reference the fact that it was originally the Reptile hood. I have increased the pixel size a few pixels vertically to allow room for the standard human heads, but it should still look good on the Lizard Heads, too.


I should easily finish this and Pirate by the end of the week.

**Comment by jrconway3** [comment]
Pirate is going to be having a lot of variants. I'm going to split off all the trim layers as well as the pirate symbols into separate overlay layers. I'll probably split this into two separate updates because there's just a ton in here and it'll make the PR bloated.

That being said, I'll be removing a lot of extra stuff, too. There's no need for male/female splits. It appears these were never updated after the repo had standardized heads added, which is why a lot of these splits still exist. So I'll be making the fixes and adjustments for the updated head positions, which will reduce a lot of assets despite me adding some more due to splitting trims into separate layers. Overall, we'll probably have a net loss of files, but only slightly.

All the Pirate/Sailor Hats will be condensed into one sheet each, per color. However, Admiral, Commodore, and Bonnie will have trim layers split off, and Bonnie/Cavalier Feather will also have the Feather split off (though this won't be a gain because we'll just replace the Feather version to ONLY have the feather).

I may organize the folder a bit more as well just to make it easier to find things...

**Comment by jrconway3** [comment]
Bonnie has a front tilt on the Feathered version when facing east... and ONLY that pose, ONLY on Feather. This... doesn't make any sense. If anything it should be consistent. Maybe I should just include both versions, one with the tilt on both west and east, and one without?

**Comment by jrconway3** [comment]
THERE'S THREE VERSIONS OF THE BONNIE SIDE HAT!

I don't... understand why this is so inconsistent?! Which is supposed to be the correct one? Ugh... I'll figure something out, but this got really complicated. The center of the hat is the same in every case, its only the edges of the hat that differ.

EDIT: I think I know where to go with it. The base Bonnie has the left side of the hat be tilted. It keeps consistent with both directions. So I'll stick with that. Its a bit confusing, but I get it now.

**Comment by jrconway3** [comment]
There were a lot more differences between the feathered and non-feathered Bonnie hats then I thought. Perhaps I went above and beyond to support two bases and the feather. I tried doing the trim, but the left-facing trim on the alt Bonnie hat is also off, so I left it out for now.


The next update should be Tricorne and Bicorne, ending the updates for all Hats. I did reorganize the hats a bit as well, though. Once this card is done I'll prepare a release on OGA for all the updated hat assets, facial accessories, etc.

**Comment by jrconway3** [comment]
Finished Tricorne hats. I'll do Bicorne tomorrow.

I really like how the trimming has been handled. I noticed that in several instances, there's a "base" hat that already exists. That happens for Thatched and Stitched, where a basic Tricorne hat exists. I finished the full sets for both pre-thatched and pre-stitched, but I'll only include the overlays here.

We also have Captain. Captain has a basic hat, one with a skull, and one with trim. The trim is its own layer now with more recolors than the original. The base had and the pirate skull are separate layers, and the skull can even be paired with the trimming.

Lieutenant is closer to the Stitched/Thatched but its slightly bigger so it doesn't seamlessly work with those assets. I have a non-trimmed version of this, and a separate trimmed layer.

**Comment by jrconway3** [comment]
I'm done now, I think. I was super tired all day so I need to run tests tomorrow before deploying the final update.

I have everything basically ready to go. Tricorne Hats are done now. I split the pirate category into Bicorne and Tricorne. The only thing I need to confirm are if the hat trim colors are the same as metal colors; if they're not then I may need to add two extra palettes there.



Kinda want to shove out the male LPC Revised clothes assets next and get that out of the way, too.

**Comment by jrconway3** [comment]
Okay, with that PR ready, I can now start working on preparing an OGA release for all hat/face assets.
